servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -276,48 +275,53 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
     if (lastInode != null) {
       if (lastInode.isDirectory() && options.isDirectory()) {
         if (!lastInode.isPersisted() && options.isPersisted()) {
-          lastInode.setPersisted(true);
-          traversalResult.getPersisted().add(lastInode);
-          String ufsPath = mMountTable.resolve(getPath(lastInode)).toString();
-          UnderFileSystem ufs = UnderFileSystem.get(ufsPath, MasterContext.getConf());
-          ufs.mkdirs(ufsPath, false);
+          // The final path component already exists and is not persisted, so it should be added
+          // to the non-persisted Inodes of traversalResult.
+          traversalResult.getNonPersisted().add(lastInode);
```

Comment:
This is a little bit confusing. Are these 2 lists supposed to be equivalent? It looks like <code>toPersistInodes</code> is initialized as <code>traversalResult.getNonPersisted()</code>. After that, I think only <code>toPersistInodes</code> should be modified and returned, but I see both of them being modified in different places. Is this intentional?